packages/data-stores/src/reader/mutations/use-site-unsubscribe-mutation.ts

```diff
@@ -28,11 +29,22 @@ const useSiteUnsubscribeMutation = () => {
 				);
 			}
 
+			let path = `/read/site/${ params.blog_id }/post_email_subscriptions/delete`;
+			let apiVersion = '1.2';
+			let body = {};
+
+			if ( isLoggedIn ) {
+				path = `/read/following/mine/delete`;
+				apiVersion = '1.1';
+				body = { source: 'calypso', url: params.url };
+			}
+
 			const response = await callApi< UnsubscribeResponse >( {
-				path: `/read/site/${ params.blog_id }/post_email_subscriptions/delete`,
+				path,
 				method: 'POST',
 				isLoggedIn,
-				apiVersion: '1.2',
+				apiVersion,
+				body,
 			} );
 			if ( ! response.success ) {
```

Comment:
As a consequence of the `/read/following/mine/delete` having a different response shape when compared to the `/read/site/${ params.blog_id }/post_email_subscriptions/delete` endpoint, the `response.success` is always `false` and hence the `Something went wrong while unsubscribing.` is thrown even when the logged-in user unsubscribed correctly.

![Markup on 2023-05-04 at 11:16:57](https://user-images.githubusercontent.com/25105483/236162341-8c92db13-b53b-47d0-a2cc-78d6a6fd7b18.png)

Reply:
I think it may be to do with the `http_envelope` argument not being appended to the request URL. Usually it is appended by our request framework based on some of the request parameters (I do not remember which ones exactly).

Reply:
BTW, I am up for a short pair coding session to help you out with this one. 🙂

Reply:
I think @ivan-ottinger is correct on this one. The endpoint is different if the user is logged in and the response is too. I didn't change the condition for that message to appear. Changing it now 👍